### DISCUSSION TITLE
How to handle fields with the same name

### DISCUSSION THREAD

**Discussion by amy-habor** [mapstruct/mapstruct#2908] [Q&A] [answered]
I have two bean
`class A {
  c_enum type,
  d_enum sub_type
}`
`class B {
  e_enum type
}`

I want to set A.sub_type value into B.type, so I write some code like this
`@Mapping(target = "type", source = "sub_type", qualifiedByName="getType")
B toB(A a);
@Named("getType")
e_enum getType(d_enum d) {
return d.toE_enum();
}`

I think it will fill b.type using getType method with source a.sub_type, but it does not work, it still uses a.type to fill b.type, and log a error

> The following constants from the property "A a.type" enum have no corresponding constant in the "B b.type" enum and must be be mapped via adding additional mappings: ...(c_enum's values).

How to fill b.type with a.sub_type instead of a.type ?

**Comment by chenzijia12300** [comment, marked as answer]
I guess this mapping statement does not execute correctly `@Mapping(target = "type", source = "sub_type", qualifiedByName="getType") B toB(A a);`, MapStruct uses the default mapping of the same name, so there may be other compilation error messages exist.

At the same time you should introduce the version you use and preferably have a demo, so that the guys can help you solve the problem faster ~

**Reply by amy-habor** [reply]
I'm sorry I have deleted those codes and forgot the exact content.
Just now I tried to reproduce this problem but failed. Maybe there is something wrong with my previous codes about `qualifiedByName` .

thank you for your help
